### DISCUSSION TITLE
to Log or not to Log transform biomass data

### DISCUSSION THREAD

**Discussion by jenniferfisher** [sdmTMB/sdmTMB#310] [Q&A] [answered]
Hi all,
I have continuous zooplankton biomass data that are very patchy in space and time with a lot of zeros. I am interested in understanding which environmental covariates (e.g., temperature, chl, aragonite) best explain the spatiotemporal patterns. I am currently using a Tweedie distribution (link = "log") in sdmTMB. If I were running a GAM (with Tweedie dist), I would normally Log10 transform the data first. Do I need to Log10 transform the data prior to using them in sdmTMB? Thanks!

**Comment by seananderson** [comment, marked as answer]
With a Tweedie distribution, you'd typically use a log link with the raw biomass data as the response. The Tweedie is for 0's and positive continuous values. With a low mean, this can create lots of zeros. I imagine you'd only log transform the response data (and add some small arbitrary value first because of the zeros) if you wanted to try to model this with a Gaussian family. Other alternatives would be `delta_gamma()` or `delta_lognormal()` within sdmTMB, which are also known as hurdle models, to model the biomass data. There's a section in the latest preprint version on these 'delta' families within sdmTMB. https://doi.org/10.1101/2022.03.24.485545

**Reply by jenniferfisher** [reply]
Hi Sean,
Thanks for replying so quickly to my transformation question. A quick follow up- I have log transformed my data and then used the Tweedie distribution in other applications because the data are still zero inflated after the transformation (see histograms below). But, it sounds like you are suggesting that the Tweedie distribution is robust to even very zero inflated data, and so it's best to stick with the raw density or biomass as the response? Thanks for your insight.

Raw density no transformation:
![image](https://github.com/pbs-assess/sdmTMB/assets/122393481/567f9564-2586-4bce-977e-cfc45cf88e4b)


Log transformed data:
![image](https://github.com/pbs-assess/sdmTMB/assets/122393481/50cea4d2-1b2e-4f9f-9b21-b672a769053b)

**Comment by seananderson** [comment]
How exactly are you getting from the first histogram to the second? The log of 0 is -Inf and the log of anything less than 1 will be negative. But, yes, the Tweedie can handle zeros just fine and the linear predictor will be in log space with a log link. After fitting you could try the `simulate()` function (`?simulate.sdmTMB`) and check the fraction of zeros in your simulated vs. fitted data to make sure they're about the same. If needed, `delta_gamma()` or `delta_lognormal()` will give more flexibility since they will separate the encounter probability (0 vs. non zero) and the positive density.

**Reply by jenniferfisher** [reply]
The second histogram is log10(density+0.1) +1 to deal with the zeros. Sounds like the Tweedie should be fine though with raw density, or a Hurdle model as you suggest- thank you.
